rules/REQUEST-920-PROTOCOL-ENFORCEMENT.conf

```diff
@@ -1080,11 +1080,26 @@ SecRule REQUEST_PROTOCOL "!@within %{tx.allowed_http_versions}" \
 # 
 # Restricted HTTP headers 
 #
-# Adding Proxy to the list as it used in a vulnerability.
-# https://access.redhat.com/security/vulnerabilities/httpoxy
+# -=[ Rule Logic ]=-
+# The use of certain headers is restricted. They are listed in the variable
+# TX.restricted_headers.
+#
+# The headers are transformed into lowercase before the match.  In order to
+# make sure that only complete header names are matching, the names in
+# TX.restricted_headers are wrapped in slashes. This guarantees that the
+# header Range (-> /range/) is not matching the restricted header
+# /content-range/ for example.
+#
+# This is a chained rule, where the first rule fills a set of variables of the
+# form TX.header_name_<HEADER_NAME>. The second rule is then executed for all
+# variables of the form TX.header_name_<HEADER_NAME>.
+#
+# As a consequence of the construction of the rule, the alert message and the
+# alert data will not display the original header name Content-Range, but
+# /content-range/ instead.
 #
-SecRule REQUEST_HEADERS_NAMES "@within %{tx.restricted_headers}" \
-  "msg:'HTTP header is restricted by policy',\
+SecRule REQUEST_HEADERS_NAMES "@rx ^(.*)$" \
```

Comment:
@dune73 do we need this capture group here? Also below, do we need to make these separate variables? can we not consistently overwrite one variable?